src/MLJInterface.jl

```diff
@@ -0,0 +1,203 @@
+        formatted.trees,
+        formatted.losses,
+        formatted.scores,
+        formatted.complexities,
+    )
+    return (;
+        best_idx=best_idx,
+        equations=formatted.trees,
+        equation_strings=equation_strings,
+        losses=formatted.losses,
+        complexities=formatted.complexities,
+        scores=formatted.scores,
+    )
+end
+
+# TODO: Pass `variable_names` and `units`
+# TODO: Enable `verbosity` being passed to `equation_search`
+MMI.clean!(::AbstractSRRegressor) = ""
+
+function MMI.fit(m::AbstractSRRegressor, verbosity, X, y, w=nothing)
+    options = get_options(m)
+    X_t = transpose(MMI.matrix(X))
+    # TODO: Is this needed? Would MLJ ever pass in a matrix for y?
+    y_t = ndims(y) == 1 ? transpose(MMI.matrix(transpose(y))) : transpose(MMI.matrix(y))
+    search_state = equation_search(
+        X_t,
+        y_t;
+        niterations=m.niterations,
+        weights=w,
+        variable_names=nothing,
+        options=options,
+        parallelism=m.parallelism,
+        numprocs=m.numprocs,
+        procs=m.procs,
+        addprocs_function=m.addprocs_function,
+        runtests=m.runtests,
+        saved_state=nothing,
+        return_state=true,
+        loss_type=m.loss_type,
+    )
+    fitresult = (; state=search_state, options=options)
+    return (fitresult, nothing, full_report(m, fitresult))
+end
+function MMI.fitted_params(m::AbstractSRRegressor, fitresult)
+    report = full_report(m, fitresult)
+    return (;
+        best_idx=report.best_idx,
+        equations=report.equations,
+        equation_strings=report.equation_strings,
+    )
+end
+function MMI.predict(m::AbstractSRRegressor, fitresult, Xnew)
+    params = MMI.fitted_params(m, fitresult)
+    Xnew_t = transpose(MMI.matrix(Xnew))
+    equations = params.equations
+    best_idx = params.best_idx
+    if isa(best_idx, Vector)
+        outs = [
+            let (out, flag) = eval_tree_array(eq[i], Xnew_t, fitresult.options)
+                !flag && error("Detected a NaN in evaluating expression.")
+                out
+            end for (i, eq) in zip(best_idx, equations)
+        ]
+        return reduce(hcat, outs)
+    else
+        out, flag = eval_tree_array(equations[best_idx], Xnew_t, fitresult.options)
+        !flag && error("Detected a NaN in evaluating expression.")
+        return out
+    end
+end
+
+MMI.package_name(::Type{<:AbstractSRRegressor}) = "SymbolicRegression"
+MMI.package_uuid(::Type{<:AbstractSRRegressor}) = "8254be44-1295-4e6a-a16d-46603ac705cb"
+function MMI.package_url(::Type{<:AbstractSRRegressor})
+    return "https://github.com/MilesCranmer/SymbolicRegression.jl"
+end
+MMI.package_license(::Type{<:AbstractSRRegressor}) = "Apache-2.0"
+MMI.is_pure_julia(::Type{<:AbstractSRRegressor}) = true
+MMI.is_wrapper(::Type{<:AbstractSRRegressor}) = false
+
+MMI.input_scitype(::Type{SRRegressor}) = MMI.Table(MMI.Continuous)
+MMI.target_scitype(::Type{SRRegressor}) = AbstractVecOrMat{<:MMI.Continuous}
```

Comment:
Is this the appropriate trait to declare to indicate that SRRegressor can handle multi-output targets?

It still seems like I get a warning if I try to fit on multi-dimensional outputs

Reply:
It's better to split your model into two.
One that handles single target. The other handles multitarget.
e.r `SRRegressor` for single target and `MultiSRRegressor` for multi-target.

Reply:
Also rather than defining these traits by hand. You can use the helper function
`metadata_model` to define it and its much cleaner see https://github.com/JuliaAI/MLJMultivariateStatsInterface.jl/blob/dev/src/models/linear_models.jl#L148

Reply:
Good idea, I will switch to that.

Reply:
Done! Added the MultiSRRegressor.

I tried out the `metadata_model` generator but it seems perhaps too abstracted away. The simple methods-based version seems a bit clearer to me. Is it okay from an implementation point of view if I stick with the methods-based version? I understand it is a bit more tedious but it is a bit more readable for me for whatever reason...